ApiTests/Tests/UI/DynamicControlsTest.cs

```diff
@@ -1,24 +1,36 @@
 ﻿using ApiTests.Tests.UI;
 using OpenQA.Selenium;
+using OpenQA.Selenium.Support.UI;
+using SeleniumExtras.WaitHelpers;
 using TestProject.Tests.Pages;
 using TestProject.Utils;
 
 namespace TestProject.Tests.UI
 {
     internal class DynamicControlsTests : BaseTest
     {
+        readonly MainPage mainPage = new MainPage();
         private static readonly By enableBtn = By.XPath(string.Format(XpathPatterns.preciseTextXpath, "Enable"));
+        private static readonly By disableBtn = By.XPath(string.Format(XpathPatterns.preciseTextXpath, "Disable"));
+        private static readonly By inputField = By.XPath(string.Format(XpathPatterns.typeText));
+        private static readonly string randomValue = Guid.NewGuid().ToString();
 
         [Test]
         public void DynamicControlsTest()
         {
-            MainPage mainPage = new MainPage();
             mainPage.ClickOnDynamicControl();
+            WebDriverWait wait = new WebDriverWait(Browser.GetDriver(), TimeSpan.FromSeconds(100));
```

Comment:
putting a random number to your code is a bad idea, you should've used the one from the config in this case. 
In case if we'd want to change the wait timeout we'd have only to change it once (in a config) instead of going through code pieces